src/ParameterLists.f90

```diff
@@ -2031,20 +2039,158 @@ SUBROUTINE getString_ParamType_scalar(thisParam,name,string,sskfmt,sdkfmt)
           TYPE IS(ParamType_SLK)
-            WRITE(tmpchar,'(i0)') param%val
+            string=str(param%val)
           TYPE IS(ParamType_SBK)
             WRITE(tmpchar,'(L1)') param%val
+            string=TRIM(ADJUSTL(tmpchar))
           TYPE IS(ParamType_STR)
-            tmpchar=param%val
+            string=param%val
+          TYPE IS(ParamType_SSK_a1)
+            WRITE(tmpchar,TRIM(sskfmtDef)) param%val(1)
+            string=TRIM(ADJUSTL(tmpchar))
+            DO i=2,SIZE(param%val)
+              WRITE(tmpchar,TRIM(sskfmtDef)) param%val(i)
+              string=string//delimDef//TRIM(ADJUSTL(tmpchar))
+            ENDDO
+          TYPE IS(ParamType_SDK_a1)
+            WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(1)
+            string=TRIM(ADJUSTL(tmpchar))
+            DO i=2,SIZE(param%val)
+              WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(i)
+              string=string//delimDef//TRIM(ADJUSTL(tmpchar))
+            ENDDO
+          TYPE IS(ParamType_SNK_a1)
+            string=str(param%val(1))
+            DO i=2,SIZE(param%val)
+              string=string//delimDef//str(param%val(i))
+            ENDDO
+          TYPE IS(ParamType_SLK_a1)
+            string=str(param%val(1))
+            DO i=2,SIZE(param%val)
+              string=string//delimDef//str(param%val(i))
+            ENDDO
+          TYPE IS(ParamType_SBK_a1)
+            WRITE(tmpchar,'(L1)') param%val(1)
+            string=TRIM(ADJUSTL(tmpchar))
+            DO i=2,SIZE(param%val)
+              WRITE(tmpchar,'(L1)') param%val(i)
+              string=string//delimDef//TRIM(ADJUSTL(tmpchar))
+            ENDDO
+          TYPE IS(ParamType_STR_a1)
+            string=''
+            string=param%val(1)
+            DO i=2,SIZE(param%val)
+              string=string//delimDef//param%val(i)
+            ENDDO
+          TYPE IS(ParamType_SSK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                WRITE(tmpchar,TRIM(sskfmtDef)) param%val(i,j)
+                string=string//TRIM(ADJUSTL(tmpchar))//delimDef
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SDK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                WRITE(tmpchar,TRIM(sdkfmtDef)) param%val(i,j)
+                string=string//TRIM(ADJUSTL(tmpchar))//delimDef
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SNK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                string=string//str(param%val(i,j))//delimDef
+              ENDDO
+            ENDDO
+          TYPE IS(ParamType_SLK_a2)
+            string=''
+            DO j=1,SIZE(param%val,DIM=2)
+              DO i=1,SIZE(param%val,DIM=1)
+                string=string//str(param%val(i,j))//delimDef
+              ENDDO
+            ENDDO
+!          TYPE IS(ParamType_SBK_a2)
```

Comment:
The logical arrays are commented out because there is no parameter list type for a 2-D array of booleans.  In the event that parameter type gets added, this code can be uncommented.  Still want to keep it?

Reply:
Probably take them out for now since they are not valid pList types yet. Looks like pretty easy to add them back if there's a need to add these types.

Reply:
Roger that.